### DISCUSSION TITLE
conditional content

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#380] [Q&A] [answered]
@pgundlach,

I would like to use conditional content, based on the value of certain elements (which would be imported from CSV [but this is a different issue]).

Given `data.xml` with the following content:

```` xml
<data>one</data>
````

How could I write a conditional in `layout.xml` that can offer the folllowing content only `if <data> == "one"`)?

```` xml
<Output>
  <Text>
    <Paragraph>
      <Value>This is one paragraph.</Value>
    </Paragraph>
  </Text>
</Output>
````

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
You can use something like this:

```xml
<Layout xmlns="urn:speedata.de:2009/publisher/en" xmlns:sd="urn:speedata:2009/publisher/functions/en">
	<Record element="data">
		<Switch>
			<Case test="not(string(.) = '')">
				<PlaceObject>
					<Textblock>
						<Paragraph>
							<Value select="." />
						</Paragraph>
					</Textblock>
				</PlaceObject>
			</Case>
		</Switch>
	</Record>
</Layout>
```

**Reply by pr-apes** [reply]
Many thanks for your reply.
